src/config-array.js

```diff
@@ -50,13 +50,19 @@ class ConfigError extends Error {
 	 * @param {number} index The index of the config object in the array.
 	 * @param {Error} source The source error. 
 	 */
-	constructor(name, index, source) {
-		super(`Config ${name}: ${source.message}`, { cause: source });
+	constructor(name, index, { cause, message }) {
+
+
+		const finalMessage = message || cause.message;
+
+		super(`Config ${name}: ${finalMessage}`, { cause });
 
 		// copy over custom properties that aren't represented
-		for (const key of Object.keys(source)) {
-			if (!(key in this)) {
-				this[key] = source[key];
+		if (cause) {
+			for (const key of Object.keys(cause)) {
+				if (!(key in this)) {
+					this[key] = cause[key];
+				}
 			}
 		}
```

Comment:
Previously, when an Error object was passed as `source`, this code copied the properties from that Error object and set that Error object as the `cause` of the ConfigError. Now it's copying the properties from `Error.cause` and setting ``Error.cause`` as the `cause` of the ConfigError, so the properties of the Error object are lost and the Error object won't be in the `cause` chain. I'm not yet sure if this is causing some observable differences in functionalities at the moment, but it might be safer to revert to the previous behavior?

Reply:
I updated the function signature but forgot to update the call sites to use the new signature. It should work the same now. (I guess this is where type checking would have been helpful!)